src/thing_creature.c

```diff
@@ -1237,6 +1270,19 @@ void terminate_thing_spell_effect(struct Thing *thing, SpellKind spkind)
         external_set_thing_state(thing, CrSt_CreatureChangeFromChicken);
         cctrl->countdown_282 = 10;
         break;
+    case SplK_Light:
+        if (thing->light_id != 0) {
+            light_delete_light(thing->light_id);
+            if (thing->alloc_flags & TAlF_IsControlled != 0)
+            {
+                create_light_for_possession(thing);
+            }
+            else
+            {
+                thing->light_id = 0;
```

Comment:
Should you delete that light with `light_delete_light` ?

Reply:
The light's already been deleted by this point.